<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="The cellxgene data portal
    (https://cellxgene.cziscience.com/) provides a graphical user
    interface to collections of single-cell sequence data processed in
    standard ways to count matrix summaries. The cellxgenedp package
    provides an alternative, R-based inteface, allowind data
    discovery, viewing, and downloading.">
<title>Discover and Access Single Cell Data Sets in the cellxgene Data Portal • cellxgenedp</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Discover and Access Single Cell Data Sets in the cellxgene Data Portal">
<meta property="og:description" content="The cellxgene data portal
    (https://cellxgene.cziscience.com/) provides a graphical user
    interface to collections of single-cell sequence data processed in
    standard ways to count matrix summaries. The cellxgenedp package
    provides an alternative, R-based inteface, allowind data
    discovery, viewing, and downloading.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">cellxgenedp</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/using_cellxgenedp.html">Discover and download datasets and files from the cellxgene data portal</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="installation-and-use">Installation and use<a class="anchor" aria-label="anchor" href="#installation-and-use"></a>
</h1></div>
<p>This package is available in <em>Bioconductor</em> version 3.15 and later. The following code installs <a href="https://bioconductor.org/packages/cellxgenedp" class="external-link">cellxgenedp</a> as well as other packages required for this vignette.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pkgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cellxgenedp"</span>, <span class="st">"zellkonverter"</span>, <span class="st">"SingleCellExperiment"</span>, <span class="st">"HDF5Array"</span><span class="op">)</span></span>
<span><span class="va">required_pkgs</span> <span class="op">&lt;-</span> <span class="va">pkgs</span><span class="op">[</span><span class="op">!</span><span class="va">pkgs</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="va">required_pkgs</span><span class="op">)</span></span></code></pre></div>
<p>Use the following <code>pkgs</code> vector to install from GitHub (latest, unchecked, development version) instead</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pkgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"mtmorgan/cellxgenedp"</span>, <span class="st">"zellkonverter"</span>, <span class="st">"SingleCellExperiment"</span>, <span class="st">"HDF5Array"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Load the package into your current <em>R</em> session. We make extensive use of the dplyr packages, and at the end of the vignette use SingleCellExperiment and zellkonverter, so load those as well.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/theislab/zellkonverter" class="external-link">zellkonverter</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span> <span class="co"># load early to avoid masking dplyr::count()</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mtmorgan.github.io/cellxgenedp/">cellxgenedp</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level1">
<h1 id="cxg-provides-a-shiny-interface">
<code>cxg()</code> Provides a ‘shiny’ interface<a class="anchor" aria-label="anchor" href="#cxg-provides-a-shiny-interface"></a>
</h1>
<p>The following sections outline how to use the <a href="https://bioconductor.org/packages/cellxgenedp" class="external-link">cellxgenedp</a> package in an <em>R</em> script; most functionality is also available in the <code><a href="reference/cxg.html">cxg()</a></code> shiny application, providing an easy way to identify, download, and visualize one or several datasets. Start the app</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/cxg.html">cxg</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>choose a project on the first tab, and a dataset for visualization, or one or more datasets for download!</p>
</div>
<div class="section level1">
<h1 id="collections-datasets-and-files">Collections, datasets and files<a class="anchor" aria-label="anchor" href="#collections-datasets-and-files"></a>
</h1>
<p>Retrieve metadata about resources available at the cellxgene data portal using <code><a href="reference/db.html">db()</a></code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">db</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/db.html">db</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Printing the <code>db</code> object provides a brief overview of the available data, as well as hints, in the form of functions like <code><a href="reference/query.html">collections()</a></code>, for further exploration.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">db</span></span>
<span><span class="co">#&gt; cellxgene_db</span></span>
<span><span class="co">#&gt; number of collections(): 119</span></span>
<span><span class="co">#&gt; number of datasets(): 706</span></span>
<span><span class="co">#&gt; number of files(): 2813</span></span></code></pre></div>
<p>The portal organizes data hierarchically, with ‘collections’ (research studies, approximately), ‘datasets’, and ‘files’. Discover data using the corresponding functions.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/query.html">collections</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 119 × 15</span></span>
<span><span class="co">#&gt;    collec…¹ acces…² conso…³ conta…⁴ conta…⁵ curat…⁶ data_…⁷ descr…⁸ links  name </span></span>
<span><span class="co">#&gt;    &lt;chr&gt;    &lt;chr&gt;   &lt;lgl&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;list&gt; &lt;chr&gt;</span></span>
<span><span class="co">#&gt;  1 43d4bb3… READ    NA      raymon… Raymon… Batuha… 1.0     Pertur… &lt;list&gt; Tran…</span></span>
<span><span class="co">#&gt;  2 0434a9d… READ    NA      avilla… Alexan… Batuha… 1.0     The va… &lt;list&gt; Acut…</span></span>
<span><span class="co">#&gt;  3 03cdc7f… READ    NA      tien.p… Tien P… Jason … 1.0     scRNA-… &lt;lgl&gt;  Emph…</span></span>
<span><span class="co">#&gt;  4 2902f08… READ    NA      lopes@… S. M. … Wei Kh… 1.0     The ov… &lt;list&gt; Sing…</span></span>
<span><span class="co">#&gt;  5 83ed3be… READ    NA      tom.ta… Tom Ta… Jennif… 1.0     During… &lt;list&gt; Inte…</span></span>
<span><span class="co">#&gt;  6 eb735cc… READ    NA      rv4@sa… Roser … Batuha… 1.0     Human … &lt;list&gt; Samp…</span></span>
<span><span class="co">#&gt;  7 44531dd… READ    NA      tallul… Tallul… Jennif… 1.0     The cr… &lt;list&gt; Sing…</span></span>
<span><span class="co">#&gt;  8 e75342a… READ    NA      nhuebn… Norber… Jennif… 1.0     Pathog… &lt;list&gt; Path…</span></span>
<span><span class="co">#&gt;  9 64b24fd… READ    NA      magnes… Scott … Rachel… 1.0     Single… &lt;list&gt; A pr…</span></span>
<span><span class="co">#&gt; 10 125eef5… READ    NA      my4@sa… Matthe… Jason … 1.0     Unders… &lt;list&gt; Sing…</span></span>
<span><span class="co">#&gt; # … with 109 more rows, 5 more variables: publisher_metadata &lt;list&gt;,</span></span>
<span><span class="co">#&gt; #   visibility &lt;chr&gt;, created_at &lt;date&gt;, published_at &lt;date&gt;,</span></span>
<span><span class="co">#&gt; #   updated_at &lt;date&gt;, and abbreviated variable names ¹​collection_id,</span></span>
<span><span class="co">#&gt; #   ²​access_type, ³​consortia, ⁴​contact_email, ⁵​contact_name, ⁶​curator_name,</span></span>
<span><span class="co">#&gt; #   ⁷​data_submission_policy_version, ⁸​description</span></span>
<span></span>
<span><span class="fu"><a href="reference/query.html">datasets</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 706 × 26</span></span>
<span><span class="co">#&gt;    dataset_id     colle…¹ donor…² assay  cell_…³ cell_…⁴ datas…⁵ devel…⁶ disease</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;          &lt;chr&gt;   &lt;list&gt;  &lt;list&gt;   &lt;int&gt; &lt;list&gt;  &lt;chr&gt;   &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  1 d62144d1-6e98… 43d4bb… &lt;list&gt;  &lt;list&gt;   68036 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  2 07f8f239-2136… 0434a9… &lt;list&gt;  &lt;list&gt;   59506 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  3 9aca7e29-d19c… 03cdc7… &lt;list&gt;  &lt;list&gt;   35699 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  4 731aeab6-e1d6… 03cdc7… &lt;list&gt;  &lt;list&gt;    3662 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  5 bfdafcbc-d785… 03cdc7… &lt;list&gt;  &lt;list&gt;   18386 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  6 218af677-b80b… 2902f0… &lt;list&gt;  &lt;list&gt;   20676 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  7 9e11bf54-0ea6… 83ed3b… &lt;list&gt;  &lt;list&gt;   71732 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  8 a0b9d32b-4085… eb735c… &lt;list&gt;  &lt;list&gt;   97499 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  9 5daeaafe-c79e… 44531d… &lt;list&gt;  &lt;list&gt;     146 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; 10 cc607851-8820… 44531d… &lt;list&gt;  &lt;list&gt;    2346 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; # … with 696 more rows, 17 more variables: is_primary_data &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   is_valid &lt;lgl&gt;, mean_genes_per_cell &lt;dbl&gt;, name &lt;chr&gt;, organism &lt;list&gt;,</span></span>
<span><span class="co">#&gt; #   processing_status &lt;list&gt;, published &lt;lgl&gt;, revision &lt;int&gt;,</span></span>
<span><span class="co">#&gt; #   schema_version &lt;chr&gt;, self_reported_ethnicity &lt;list&gt;, sex &lt;list&gt;,</span></span>
<span><span class="co">#&gt; #   suspension_type &lt;list&gt;, tissue &lt;list&gt;, tombstone &lt;lgl&gt;, created_at &lt;date&gt;,</span></span>
<span><span class="co">#&gt; #   published_at &lt;date&gt;, updated_at &lt;date&gt;, and abbreviated variable names</span></span>
<span><span class="co">#&gt; #   ¹​collection_id, ²​donor_id, ³​cell_count, ⁴​cell_type, ⁵​dataset_deployments, …</span></span>
<span></span>
<span><span class="fu"><a href="reference/query.html">files</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 2,813 × 8</span></span>
<span><span class="co">#&gt;    file_id          datas…¹ filen…² filet…³ s3_uri user_…⁴ created_at updated_at</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;            &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;lgl&gt;   &lt;date&gt;     &lt;date&gt;    </span></span>
<span><span class="co">#&gt;  1 87ce2086-f2eb-4… d62144… "raw.h… RAW_H5… s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  2 8eabe485-be71-4… d62144… "local… H5AD    s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  3 33dfa406-132a-4… d62144… "local… RDS     s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  4 454e0b3c-e207-4… d62144… ""      CXG     s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  5 62deea2a-d722-4… 07f8f2… "raw.h… RAW_H5… s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  6 b50c48f1-ef5e-4… 07f8f2… "local… H5AD    s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  7 59309f21-0db8-4… 07f8f2… "local… RDS     s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  8 399c2655-f56a-4… 07f8f2… ""      CXG     s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  9 1ca25a06-7251-4… 9aca7e… "raw.h… RAW_H5… s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt; 10 a2dd5b48-98df-4… 9aca7e… "local… H5AD    s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt; # … with 2,803 more rows, and abbreviated variable names ¹​dataset_id,</span></span>
<span><span class="co">#&gt; #   ²​filename, ³​filetype, ⁴​user_submitted</span></span></code></pre></div>
<p>Each of these resources has a unique primary identifier (e.g., <code>file_id</code>) as well as an identifier describing the relationship of the resource to other components of the database (e.g., <code>dataset_id</code>). These identifiers can be used to ‘join’ information across tables.</p>
<div class="section level2">
<h2 id="using-dplyr-to-navigate-data">Using <code>dplyr</code> to navigate data<a class="anchor" aria-label="anchor" href="#using-dplyr-to-navigate-data"></a>
</h2>
<p>A collection may have several datasets, and datasets may have several files. For instance, here is the collection with the most datasets</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">collection_with_most_datasets</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="reference/query.html">datasets</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">collection_id</span>, sort <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>We can find out about this collection by joining with the <code><a href="reference/query.html">collections()</a></code> table.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>    <span class="va">collection_with_most_datasets</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">collection_id</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="reference/query.html">collections</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span>,</span>
<span>    by <span class="op">=</span> <span class="st">"collection_id"</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 1</span></span>
<span><span class="co">#&gt; Columns: 15</span></span>
<span><span class="co">#&gt; $ collection_id                  &lt;chr&gt; "283d65eb-dd53-496d-adb7-7570c7caa443"</span></span>
<span><span class="co">#&gt; $ access_type                    &lt;chr&gt; "READ"</span></span>
<span><span class="co">#&gt; $ consortia                      &lt;lgl&gt; NA</span></span>
<span><span class="co">#&gt; $ contact_email                  &lt;chr&gt; "kimberly.siletti@ki.se"</span></span>
<span><span class="co">#&gt; $ contact_name                   &lt;chr&gt; "Kimberly Siletti"</span></span>
<span><span class="co">#&gt; $ curator_name                   &lt;chr&gt; "James Chaffer"</span></span>
<span><span class="co">#&gt; $ data_submission_policy_version &lt;chr&gt; "1.0"</span></span>
<span><span class="co">#&gt; $ description                    &lt;chr&gt; "The human brain directs a wide range o…</span></span>
<span><span class="co">#&gt; $ links                          &lt;list&gt; [["", "RAW_DATA", "http://data.nemoarch…</span></span>
<span><span class="co">#&gt; $ name                           &lt;chr&gt; "Transcriptomic diversity of cell types…</span></span>
<span><span class="co">#&gt; $ publisher_metadata             &lt;list&gt; [[["Siletti", "Kimberly"], ["Hodge", "R…</span></span>
<span><span class="co">#&gt; $ visibility                     &lt;chr&gt; "PUBLIC"</span></span>
<span><span class="co">#&gt; $ created_at                     &lt;date&gt; 2022-11-09</span></span>
<span><span class="co">#&gt; $ published_at                   &lt;date&gt; 2022-12-09</span></span>
<span><span class="co">#&gt; $ updated_at                     &lt;date&gt; 2022-12-09</span></span></code></pre></div>
<p>We can take a similar strategy to identify all datasets belonging to this collection</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>    <span class="va">collection_with_most_datasets</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">collection_id</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="reference/query.html">datasets</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span>,</span>
<span>    by <span class="op">=</span> <span class="st">"collection_id"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in left_join(select(collection_with_most_datasets, collection_id), : Each row in `x` is expected to match at most 1 row in `y`.</span></span>
<span><span class="co">#&gt; ℹ Row 1 of `x` matches multiple rows.</span></span>
<span><span class="co">#&gt; ℹ If multiple matches are expected, set `multiple = "all"` to silence this</span></span>
<span><span class="co">#&gt;   warning.</span></span>
<span><span class="co">#&gt; # A tibble: 138 × 26</span></span>
<span><span class="co">#&gt;    collection_id  datas…¹ donor…² assay  cell_…³ cell_…⁴ datas…⁵ devel…⁶ disease</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;          &lt;chr&gt;   &lt;list&gt;  &lt;list&gt;   &lt;int&gt; &lt;list&gt;  &lt;chr&gt;   &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  1 283d65eb-dd53… 4fb0cd… &lt;list&gt;  &lt;list&gt;   25071 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  2 283d65eb-dd53… d15266… &lt;list&gt;  &lt;list&gt;   35290 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  3 283d65eb-dd53… f4c89e… &lt;list&gt;  &lt;list&gt;   38331 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  4 283d65eb-dd53… fb3dee… &lt;list&gt;  &lt;list&gt;   21534 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  5 283d65eb-dd53… 7e0f26… &lt;list&gt;  &lt;list&gt;    9932 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  6 283d65eb-dd53… 0834c8… &lt;list&gt;  &lt;list&gt;   28724 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  7 283d65eb-dd53… 4cce2e… &lt;list&gt;  &lt;list&gt;   35359 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  8 283d65eb-dd53… 05dd9b… &lt;list&gt;  &lt;list&gt;   27210 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt;  9 283d65eb-dd53… 5b994d… &lt;list&gt;  &lt;list&gt;   26172 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; 10 283d65eb-dd53… dab53e… &lt;list&gt;  &lt;list&gt;   34933 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; # … with 128 more rows, 17 more variables: is_primary_data &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   is_valid &lt;lgl&gt;, mean_genes_per_cell &lt;dbl&gt;, name &lt;chr&gt;, organism &lt;list&gt;,</span></span>
<span><span class="co">#&gt; #   processing_status &lt;list&gt;, published &lt;lgl&gt;, revision &lt;int&gt;,</span></span>
<span><span class="co">#&gt; #   schema_version &lt;chr&gt;, self_reported_ethnicity &lt;list&gt;, sex &lt;list&gt;,</span></span>
<span><span class="co">#&gt; #   suspension_type &lt;list&gt;, tissue &lt;list&gt;, tombstone &lt;lgl&gt;, created_at &lt;date&gt;,</span></span>
<span><span class="co">#&gt; #   published_at &lt;date&gt;, updated_at &lt;date&gt;, and abbreviated variable names</span></span>
<span><span class="co">#&gt; #   ¹​dataset_id, ²​donor_id, ³​cell_count, ⁴​cell_type, ⁵​dataset_deployments, …</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="facets-provides-information-on-levels-present-in-specific-columns">
<code>facets()</code> provides information on ‘levels’ present in specific columns<a class="anchor" aria-label="anchor" href="#facets-provides-information-on-levels-present-in-specific-columns"></a>
</h2>
<p>Notice that some columns are ‘lists’ rather than atomic vectors like ‘character’ or ‘integer’.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/query.html">datasets</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/where.html" class="external-link">where</a></span><span class="op">(</span><span class="va">is.list</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 706 × 11</span></span>
<span><span class="co">#&gt;    donor_id   assay  cell_…¹ devel…² disease organ…³ processing…⁴ self_…⁵ sex   </span></span>
<span><span class="co">#&gt;    &lt;list&gt;     &lt;list&gt; &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;list&gt;       &lt;list&gt;  &lt;list&gt;</span></span>
<span><span class="co">#&gt;  1 &lt;list [9]&gt; &lt;list&gt; &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;named list&gt; &lt;list&gt;  &lt;list&gt;</span></span>
<span><span class="co">#&gt;  2 &lt;list&gt;     &lt;list&gt; &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;named list&gt; &lt;list&gt;  &lt;list&gt;</span></span>
<span><span class="co">#&gt;  3 &lt;list [6]&gt; &lt;list&gt; &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;named list&gt; &lt;list&gt;  &lt;list&gt;</span></span>
<span><span class="co">#&gt;  4 &lt;list [6]&gt; &lt;list&gt; &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;named list&gt; &lt;list&gt;  &lt;list&gt;</span></span>
<span><span class="co">#&gt;  5 &lt;list [6]&gt; &lt;list&gt; &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;named list&gt; &lt;list&gt;  &lt;list&gt;</span></span>
<span><span class="co">#&gt;  6 &lt;list [5]&gt; &lt;list&gt; &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;named list&gt; &lt;list&gt;  &lt;list&gt;</span></span>
<span><span class="co">#&gt;  7 &lt;list [5]&gt; &lt;list&gt; &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;named list&gt; &lt;list&gt;  &lt;list&gt;</span></span>
<span><span class="co">#&gt;  8 &lt;list&gt;     &lt;list&gt; &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;named list&gt; &lt;list&gt;  &lt;list&gt;</span></span>
<span><span class="co">#&gt;  9 &lt;list [4]&gt; &lt;list&gt; &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;named list&gt; &lt;list&gt;  &lt;list&gt;</span></span>
<span><span class="co">#&gt; 10 &lt;list [4]&gt; &lt;list&gt; &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;list&gt;  &lt;named list&gt; &lt;list&gt;  &lt;list&gt;</span></span>
<span><span class="co">#&gt; # … with 696 more rows, 2 more variables: suspension_type &lt;list&gt;,</span></span>
<span><span class="co">#&gt; #   tissue &lt;list&gt;, and abbreviated variable names ¹​cell_type,</span></span>
<span><span class="co">#&gt; #   ²​development_stage, ³​organism, ⁴​processing_status, ⁵​self_reported_ethnicity</span></span></code></pre></div>
<p>This indicates that at least some of the datasets had more than one type of <code>assay</code>, <code>cell_type</code>, etc. The <code><a href="reference/facets.html">facets()</a></code> function provides a convenient way of discovering possible levels of each column, e.g., <code>assay</code>, <code>organism</code>, <code>self_reported_ethnicity</code>, or <code>sex</code>, and the number of datasets with each label.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/facets.html">facets</a></span><span class="op">(</span><span class="va">db</span>, <span class="st">"assay"</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 32 × 4</span></span>
<span><span class="co">#&gt;    facet label                          ontology_term_id     n</span></span>
<span><span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt;                          &lt;chr&gt;            &lt;int&gt;</span></span>
<span><span class="co">#&gt;  1 assay 10x 3' v3                      EFO:0009922        346</span></span>
<span><span class="co">#&gt;  2 assay 10x 3' v2                      EFO:0009899        170</span></span>
<span><span class="co">#&gt;  3 assay Slide-seqV2                    EFO:0030062        129</span></span>
<span><span class="co">#&gt;  4 assay 10x 5' v1                      EFO:0011025         48</span></span>
<span><span class="co">#&gt;  5 assay Smart-seq2                     EFO:0008931         38</span></span>
<span><span class="co">#&gt;  6 assay Visium Spatial Gene Expression EFO:0010961         35</span></span>
<span><span class="co">#&gt;  7 assay 10x multiome                   EFO:0030059         24</span></span>
<span><span class="co">#&gt;  8 assay Drop-seq                       EFO:0008722         12</span></span>
<span><span class="co">#&gt;  9 assay 10x 3' transcription profiling EFO:0030003          9</span></span>
<span><span class="co">#&gt; 10 assay 10x 5' v2                      EFO:0009900          9</span></span>
<span><span class="co">#&gt; # … with 22 more rows</span></span>
<span><span class="fu"><a href="reference/facets.html">facets</a></span><span class="op">(</span><span class="va">db</span>, <span class="st">"self_reported_ethnicity"</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 18 × 4</span></span>
<span><span class="co">#&gt;    facet                   label                                   ontol…¹     n</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;                   &lt;chr&gt;                                   &lt;chr&gt;   &lt;int&gt;</span></span>
<span><span class="co">#&gt;  1 self_reported_ethnicity unknown                                 unknown   358</span></span>
<span><span class="co">#&gt;  2 self_reported_ethnicity European                                HANCES…   200</span></span>
<span><span class="co">#&gt;  3 self_reported_ethnicity na                                      na        184</span></span>
<span><span class="co">#&gt;  4 self_reported_ethnicity Asian                                   HANCES…    81</span></span>
<span><span class="co">#&gt;  5 self_reported_ethnicity African American                        HANCES…    39</span></span>
<span><span class="co">#&gt;  6 self_reported_ethnicity multiethnic                             multie…    25</span></span>
<span><span class="co">#&gt;  7 self_reported_ethnicity Greater Middle Eastern  (Middle Easter… HANCES…    21</span></span>
<span><span class="co">#&gt;  8 self_reported_ethnicity Hispanic or Latin American              HANCES…    16</span></span>
<span><span class="co">#&gt;  9 self_reported_ethnicity African American or Afro-Caribbean      HANCES…    10</span></span>
<span><span class="co">#&gt; 10 self_reported_ethnicity East Asian                              HANCES…     4</span></span>
<span><span class="co">#&gt; 11 self_reported_ethnicity African                                 HANCES…     3</span></span>
<span><span class="co">#&gt; 12 self_reported_ethnicity South Asian                             HANCES…     2</span></span>
<span><span class="co">#&gt; 13 self_reported_ethnicity Chinese                                 HANCES…     1</span></span>
<span><span class="co">#&gt; 14 self_reported_ethnicity Eskimo                                  HANCES…     1</span></span>
<span><span class="co">#&gt; 15 self_reported_ethnicity Finnish                                 HANCES…     1</span></span>
<span><span class="co">#&gt; 16 self_reported_ethnicity Han Chinese                             HANCES…     1</span></span>
<span><span class="co">#&gt; 17 self_reported_ethnicity Oceanian                                HANCES…     1</span></span>
<span><span class="co">#&gt; 18 self_reported_ethnicity Pacific Islander                        HANCES…     1</span></span>
<span><span class="co">#&gt; # … with abbreviated variable name ¹​ontology_term_id</span></span>
<span><span class="fu"><a href="reference/facets.html">facets</a></span><span class="op">(</span><span class="va">db</span>, <span class="st">"sex"</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 3 × 4</span></span>
<span><span class="co">#&gt;   facet label   ontology_term_id     n</span></span>
<span><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;   &lt;chr&gt;            &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1 sex   male    PATO:0000384       614</span></span>
<span><span class="co">#&gt; 2 sex   female  PATO:0000383       371</span></span>
<span><span class="co">#&gt; 3 sex   unknown unknown             54</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="filtering-faceted-columns">Filtering faceted columns<a class="anchor" aria-label="anchor" href="#filtering-faceted-columns"></a>
</h2>
<p>Suppose we were interested in finding datasets from the 10x 3’ v3 assay (<code>ontology_term_id</code> of <code>EFO:0009922</code>) containing individuals of African American ethnicity, and female sex. Use the <code><a href="reference/facets.html">facets_filter()</a></code> utility function to filter data sets as needed</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">african_american_female</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="reference/query.html">datasets</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="reference/facets.html">facets_filter</a></span><span class="op">(</span><span class="va">assay</span>, <span class="st">"ontology_term_id"</span>, <span class="st">"EFO:0009922"</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="reference/facets.html">facets_filter</a></span><span class="op">(</span><span class="va">self_reported_ethnicity</span>, <span class="st">"label"</span>, <span class="st">"African American"</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="reference/facets.html">facets_filter</a></span><span class="op">(</span><span class="va">sex</span>, <span class="st">"label"</span>, <span class="st">"female"</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>Use <code>nrow(african_american_female)</code> to find the number of datasets satisfying our criteria. It looks like there are up to</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">african_american_female</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>total_cell_count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">cell_count</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 1 × 1</span></span>
<span><span class="co">#&gt;   total_cell_count</span></span>
<span><span class="co">#&gt;              &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1          2657433</span></span></code></pre></div>
<p>cells sequenced (each dataset may contain cells from several ethnicities, as well as males or individuals of unknown gender, so we do not know the actual number of cells available without downloading files). Use <code>left_join</code> to identify the corresponding collections:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## collections</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>    <span class="va">african_american_female</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">collection_id</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="reference/query.html">collections</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span>,</span>
<span>    by <span class="op">=</span> <span class="st">"collection_id"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 8 × 15</span></span>
<span><span class="co">#&gt;   collect…¹ acces…² conso…³ conta…⁴ conta…⁵ curat…⁶ data_…⁷ descr…⁸ links  name </span></span>
<span><span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;   &lt;lgl&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;list&gt; &lt;chr&gt;</span></span>
<span><span class="co">#&gt; 1 c9706a92… READ    NA      hnaksh… Harikr… Jennif… 1.0     "Singl… &lt;list&gt; A si…</span></span>
<span><span class="co">#&gt; 2 2f75d249… READ    NA      rsatij… Rahul … Jennif… 1.0     "This … &lt;list&gt; Azim…</span></span>
<span><span class="co">#&gt; 3 62e8f058… READ    NA      chanj3… Joseph… Jennif… 1.0     "155,0… &lt;list&gt; HTAN…</span></span>
<span><span class="co">#&gt; 4 625f6bf4… READ    NA      a5wang… Allen … Jennif… 1.0     "Large… &lt;list&gt; Lung…</span></span>
<span><span class="co">#&gt; 5 b953c942… READ    NA      icobos… Inma C… Jennif… 1.0     "Tau a… &lt;list&gt; Sing…</span></span>
<span><span class="co">#&gt; 6 bcb61471… READ    NA      info@k… KPMP    Jennif… 1.0     "Under… &lt;list&gt; An a…</span></span>
<span><span class="co">#&gt; 7 b9fc3d70… READ    NA      bruce.… Bruce … Jennif… 1.0     "Numer… &lt;list&gt; A We…</span></span>
<span><span class="co">#&gt; 8 a98b828a… READ    NA      markus… Markus… Jennif… 1.0     "Tumor… &lt;list&gt; HCA …</span></span>
<span><span class="co">#&gt; # … with 5 more variables: publisher_metadata &lt;list&gt;, visibility &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   created_at &lt;date&gt;, published_at &lt;date&gt;, updated_at &lt;date&gt;, and abbreviated</span></span>
<span><span class="co">#&gt; #   variable names ¹​collection_id, ²​access_type, ³​consortia, ⁴​contact_email,</span></span>
<span><span class="co">#&gt; #   ⁵​contact_name, ⁶​curator_name, ⁷​data_submission_policy_version, ⁸​description</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="publication-and-other-external-data">Publication and other external data<a class="anchor" aria-label="anchor" href="#publication-and-other-external-data"></a>
</h2>
<p>Many collections include publication information and other external data. This information is available in the return value of <code><a href="reference/query.html">collections()</a></code>, but the helper function <code><a href="reference/query.html">publisher_metadata()</a></code>, <code><a href="reference/query.html">authors()</a></code>, and <code><a href="reference/query.html">links()</a></code> may facilite access.</p>
<p>Suppose one is interested in the publication “A single-cell atlas of the healthy breast tissues reveals clinically relevant clusters of breast epithelial cells”. Discover it in the collections</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">title_of_interest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span></span>
<span>    <span class="st">"A single-cell atlas of the healthy breast tissues reveals clinically"</span>,</span>
<span>    <span class="st">"relevant clusters of breast epithelial cells"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">collection_of_interest</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="reference/query.html">collections</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/startsWith.html" class="external-link">startsWith</a></span><span class="op">(</span><span class="va">name</span>, <span class="va">title_of_interest</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">collection_of_interest</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 1</span></span>
<span><span class="co">#&gt; Columns: 15</span></span>
<span><span class="co">#&gt; $ collection_id                  &lt;chr&gt; "c9706a92-0e5f-46c1-96d8-20e42467f287"</span></span>
<span><span class="co">#&gt; $ access_type                    &lt;chr&gt; "READ"</span></span>
<span><span class="co">#&gt; $ consortia                      &lt;lgl&gt; NA</span></span>
<span><span class="co">#&gt; $ contact_email                  &lt;chr&gt; "hnakshat@iupui.edu"</span></span>
<span><span class="co">#&gt; $ contact_name                   &lt;chr&gt; "Harikrishna Nakshatri"</span></span>
<span><span class="co">#&gt; $ curator_name                   &lt;chr&gt; "Jennifer Yu-Sheng Chien"</span></span>
<span><span class="co">#&gt; $ data_submission_policy_version &lt;chr&gt; "1.0"</span></span>
<span><span class="co">#&gt; $ description                    &lt;chr&gt; "Single-cell RNA sequencing (scRNA-seq)…</span></span>
<span><span class="co">#&gt; $ links                          &lt;list&gt; [["", "RAW_DATA", "https://data.humance…</span></span>
<span><span class="co">#&gt; $ name                           &lt;chr&gt; "A single-cell atlas of the healthy bre…</span></span>
<span><span class="co">#&gt; $ publisher_metadata             &lt;list&gt; [[["Bhat-Nakshatri", "Poornima"], ["Gao…</span></span>
<span><span class="co">#&gt; $ visibility                     &lt;chr&gt; "PUBLIC"</span></span>
<span><span class="co">#&gt; $ created_at                     &lt;date&gt; 2021-03-25</span></span>
<span><span class="co">#&gt; $ published_at                   &lt;date&gt; 2021-03-25</span></span>
<span><span class="co">#&gt; $ updated_at                     &lt;date&gt; 2021-03-25</span></span></code></pre></div>
<p>Use the <code>collection_id</code> to extract publisher metadata (including a DOI if available) and author information</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">collection_id_of_interest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">collection_of_interest</span>, <span class="st">"collection_id"</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/query.html">publisher_metadata</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">collection_id</span> <span class="op">==</span> <span class="va">collection_id_of_interest</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 1</span></span>
<span><span class="co">#&gt; Columns: 9</span></span>
<span><span class="co">#&gt; $ collection_id   &lt;chr&gt; "c9706a92-0e5f-46c1-96d8-20e42467f287"</span></span>
<span><span class="co">#&gt; $ name            &lt;chr&gt; "A single-cell atlas of the healthy breast tissues rev…</span></span>
<span><span class="co">#&gt; $ is_preprint     &lt;lgl&gt; FALSE</span></span>
<span><span class="co">#&gt; $ journal         &lt;chr&gt; "Cell Reports Medicine"</span></span>
<span><span class="co">#&gt; $ published_at    &lt;date&gt; 2021-03-01</span></span>
<span><span class="co">#&gt; $ published_year  &lt;int&gt; 2021</span></span>
<span><span class="co">#&gt; $ published_month &lt;int&gt; 3</span></span>
<span><span class="co">#&gt; $ published_day   &lt;int&gt; 1</span></span>
<span><span class="co">#&gt; $ doi             &lt;chr&gt; "https://doi.org/10.1016/j.xcrm.2021.100219"</span></span>
<span><span class="fu"><a href="reference/query.html">authors</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">collection_id</span> <span class="op">==</span> <span class="va">collection_id_of_interest</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 12 × 4</span></span>
<span><span class="co">#&gt;    collection_id                        family         given       consortium</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;                                &lt;chr&gt;          &lt;chr&gt;       &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 c9706a92-0e5f-46c1-96d8-20e42467f287 Bhat-Nakshatri Poornima    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  2 c9706a92-0e5f-46c1-96d8-20e42467f287 Gao            Hongyu      &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  3 c9706a92-0e5f-46c1-96d8-20e42467f287 Sheng          Liu         &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  4 c9706a92-0e5f-46c1-96d8-20e42467f287 McGuire        Patrick C.  &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  5 c9706a92-0e5f-46c1-96d8-20e42467f287 Xuei           Xiaoling    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  6 c9706a92-0e5f-46c1-96d8-20e42467f287 Wan            Jun         &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  7 c9706a92-0e5f-46c1-96d8-20e42467f287 Liu            Yunlong     &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  8 c9706a92-0e5f-46c1-96d8-20e42467f287 Althouse       Sandra K.   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  9 c9706a92-0e5f-46c1-96d8-20e42467f287 Colter         Austyn      &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 c9706a92-0e5f-46c1-96d8-20e42467f287 Sandusky       George      &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 11 c9706a92-0e5f-46c1-96d8-20e42467f287 Storniolo      Anna Maria  &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 12 c9706a92-0e5f-46c1-96d8-20e42467f287 Nakshatri      Harikrishna &lt;NA&gt;</span></span></code></pre></div>
<p>Collections may have links to additional external data, in this case a DOI and two links to <code>RAW_DATA</code>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">external_links</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/query.html">links</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span></span>
<span><span class="va">external_links</span></span>
<span><span class="co">#&gt; # A tibble: 586 × 4</span></span>
<span><span class="co">#&gt;    collection_id                        link_name                link_…¹ link_…²</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;                                &lt;chr&gt;                    &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt;  1 43d4bb39-21af-4d05-b973-4c1fed7b916c &lt;NA&gt;                     OTHER   http:/…</span></span>
<span><span class="co">#&gt;  2 43d4bb39-21af-4d05-b973-4c1fed7b916c &lt;NA&gt;                     DOI     https:…</span></span>
<span><span class="co">#&gt;  3 43d4bb39-21af-4d05-b973-4c1fed7b916c EGAS00001002927          DATA_S… https:…</span></span>
<span><span class="co">#&gt;  4 0434a9d4-85fd-4554-b8e3-cf6c582bb2fa Purification of PBMCs    PROTOC… https:…</span></span>
<span><span class="co">#&gt;  5 0434a9d4-85fd-4554-b8e3-cf6c582bb2fa &lt;NA&gt;                     DOI     https:…</span></span>
<span><span class="co">#&gt;  6 0434a9d4-85fd-4554-b8e3-cf6c582bb2fa MGH COVID-19 Effort Blo… PROTOC… http:/…</span></span>
<span><span class="co">#&gt;  7 0434a9d4-85fd-4554-b8e3-cf6c582bb2fa COVID Blood Processing … PROTOC… http:/…</span></span>
<span><span class="co">#&gt;  8 0434a9d4-85fd-4554-b8e3-cf6c582bb2fa CITE-seq for PBMCs       PROTOC… https:…</span></span>
<span><span class="co">#&gt;  9 0434a9d4-85fd-4554-b8e3-cf6c582bb2fa COVID Airway Processing… PROTOC… http:/…</span></span>
<span><span class="co">#&gt; 10 0434a9d4-85fd-4554-b8e3-cf6c582bb2fa &lt;NA&gt;                     OTHER   https:…</span></span>
<span><span class="co">#&gt; # … with 576 more rows, and abbreviated variable names ¹​link_type, ²​link_url</span></span>
<span><span class="va">external_links</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">link_type</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 2</span></span>
<span><span class="co">#&gt;   link_type       n</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;       &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1 DATA_SOURCE    37</span></span>
<span><span class="co">#&gt; 2 DOI           115</span></span>
<span><span class="co">#&gt; 3 LAB_WEBSITE    31</span></span>
<span><span class="co">#&gt; 4 OTHER         205</span></span>
<span><span class="co">#&gt; 5 PROTOCOL       36</span></span>
<span><span class="co">#&gt; 6 RAW_DATA      162</span></span>
<span><span class="va">external_links</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">collection_id</span> <span class="op">==</span> <span class="va">collection_id_of_interest</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 3 × 4</span></span>
<span><span class="co">#&gt;   collection_id                        link_name link_type link_url             </span></span>
<span><span class="co">#&gt;   &lt;chr&gt;                                &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;                </span></span>
<span><span class="co">#&gt; 1 c9706a92-0e5f-46c1-96d8-20e42467f287 &lt;NA&gt;      RAW_DATA  https://data.humance…</span></span>
<span><span class="co">#&gt; 2 c9706a92-0e5f-46c1-96d8-20e42467f287 &lt;NA&gt;      DOI       https://doi.org/10.1…</span></span>
<span><span class="co">#&gt; 3 c9706a92-0e5f-46c1-96d8-20e42467f287 &lt;NA&gt;      RAW_DATA  https://www.ncbi.nlm…</span></span></code></pre></div>
<p>Conversely, knowledge of a DOI, etc., can be used to discover details of the corresponding collection.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">doi_of_interest</span> <span class="op">&lt;-</span> <span class="st">"https://doi.org/10.1016/j.xcrm.2021.100219"</span></span>
<span><span class="fu"><a href="reference/query.html">links</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">link_type</span> <span class="op">==</span> <span class="st">"DOI"</span> <span class="op">&amp;</span> <span class="va">link_url</span> <span class="op">==</span> <span class="va">doi_of_interest</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="fu"><a href="reference/query.html">collections</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(collection_id)`</span></span>
<span><span class="co">#&gt; Rows: 1</span></span>
<span><span class="co">#&gt; Columns: 18</span></span>
<span><span class="co">#&gt; $ collection_id                  &lt;chr&gt; "c9706a92-0e5f-46c1-96d8-20e42467f287"</span></span>
<span><span class="co">#&gt; $ link_name                      &lt;chr&gt; NA</span></span>
<span><span class="co">#&gt; $ link_type                      &lt;chr&gt; "DOI"</span></span>
<span><span class="co">#&gt; $ link_url                       &lt;chr&gt; "https://doi.org/10.1016/j.xcrm.2021.10…</span></span>
<span><span class="co">#&gt; $ access_type                    &lt;chr&gt; "READ"</span></span>
<span><span class="co">#&gt; $ consortia                      &lt;lgl&gt; NA</span></span>
<span><span class="co">#&gt; $ contact_email                  &lt;chr&gt; "hnakshat@iupui.edu"</span></span>
<span><span class="co">#&gt; $ contact_name                   &lt;chr&gt; "Harikrishna Nakshatri"</span></span>
<span><span class="co">#&gt; $ curator_name                   &lt;chr&gt; "Jennifer Yu-Sheng Chien"</span></span>
<span><span class="co">#&gt; $ data_submission_policy_version &lt;chr&gt; "1.0"</span></span>
<span><span class="co">#&gt; $ description                    &lt;chr&gt; "Single-cell RNA sequencing (scRNA-seq)…</span></span>
<span><span class="co">#&gt; $ links                          &lt;list&gt; [["", "RAW_DATA", "https://data.humance…</span></span>
<span><span class="co">#&gt; $ name                           &lt;chr&gt; "A single-cell atlas of the healthy bre…</span></span>
<span><span class="co">#&gt; $ publisher_metadata             &lt;list&gt; [[["Bhat-Nakshatri", "Poornima"], ["Gao…</span></span>
<span><span class="co">#&gt; $ visibility                     &lt;chr&gt; "PUBLIC"</span></span>
<span><span class="co">#&gt; $ created_at                     &lt;date&gt; 2021-03-25</span></span>
<span><span class="co">#&gt; $ published_at                   &lt;date&gt; 2021-03-25</span></span>
<span><span class="co">#&gt; $ updated_at                     &lt;date&gt; 2021-03-25</span></span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="visualizing-data-in-cellxgene">Visualizing data in <code>cellxgene</code>
<a class="anchor" aria-label="anchor" href="#visualizing-data-in-cellxgene"></a>
</h1>
<p>Discover files associated with our first selected dataset</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">selected_files</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>        <span class="va">african_american_female</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">dataset_id</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="reference/query.html">files</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span>,</span>
<span>        by <span class="op">=</span> <span class="st">"dataset_id"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="co">#&gt; Warning in left_join(select(african_american_female, dataset_id), files(db), : Each row in `x` is expected to match at most 1 row in `y`.</span></span>
<span><span class="co">#&gt; ℹ Row 1 of `x` matches multiple rows.</span></span>
<span><span class="co">#&gt; ℹ If multiple matches are expected, set `multiple = "all"` to silence this</span></span>
<span><span class="co">#&gt;   warning.</span></span>
<span><span class="va">selected_files</span></span>
<span><span class="co">#&gt; # A tibble: 88 × 8</span></span>
<span><span class="co">#&gt;    dataset_id       file_id filen…¹ filet…² s3_uri user_…³ created_at updated_at</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;            &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;lgl&gt;   &lt;date&gt;     &lt;date&gt;    </span></span>
<span><span class="co">#&gt;  1 24205601-0780-4… e1c842… "raw.h… RAW_H5… s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  2 24205601-0780-4… 15e9d9… "local… H5AD    s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  3 24205601-0780-4… 0d3974… ""      CXG     s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  4 24205601-0780-4… e254f9… "local… RDS     s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  5 d084f1e1-42f1-4… 59bd46… "local… RDS     s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  6 d084f1e1-42f1-4… 3a2467… ""      CXG     s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  7 d084f1e1-42f1-4… d9f9d0… "local… H5AD    s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  8 d084f1e1-42f1-4… bee803… "raw.h… RAW_H5… s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt;  9 3f40e6b1-e280-4… e47e72… "raw.h… RAW_H5… s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt; 10 3f40e6b1-e280-4… d72756… "local… H5AD    s3://… TRUE    1970-01-01 1970-01-01</span></span>
<span><span class="co">#&gt; # … with 78 more rows, and abbreviated variable names ¹​filename, ²​filetype,</span></span>
<span><span class="co">#&gt; #   ³​user_submitted</span></span></code></pre></div>
<p>The <code>filetype</code> column lists the type of each file. The cellxgene service can be used to visualize <em>datasets</em> that have <code>CXG</code> files.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">selected_files</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">filetype</span> <span class="op">==</span> <span class="st">"CXG"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="co"># visualize a single dataset</span></span>
<span>    <span class="fu"><a href="reference/query.html">datasets_visualize</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Visualization is an interactive process, so <code><a href="reference/query.html">datasets_visualize()</a></code> will only open up to 5 browser tabs per call.</p>
</div>
<div class="section level1">
<h1 id="file-download-and-use">File download and use<a class="anchor" aria-label="anchor" href="#file-download-and-use"></a>
</h1>
<p>Datasets usually contain <code>CXG</code> (cellxgene visualization), <code>H5AD</code> (files produced by the python AnnData module), and <code>Rds</code> (serialized files produced by the <em>R</em> Seurat package). There are no public parsers for <code>CXG</code>, and the <code>Rds</code> files may be unreadable if the version of Seurat used to create the file is different from the version used to read the file. We therefore focus on the <code>H5AD</code> files. For illustration, we download one of our selected files.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">local_file</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">selected_files</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>        <span class="va">dataset_id</span> <span class="op">==</span> <span class="st">"24205601-0780-4bf2-b1d9-0e3cacbc2cd6"</span>,</span>
<span>        <span class="va">filetype</span> <span class="op">==</span> <span class="st">"H5AD"</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="reference/query.html">files_download</a></span><span class="op">(</span>dry.run <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">local_file</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "15e9d9af-60dc-4897-88e4-89842655d6b8.H5AD"</span></span></code></pre></div>
<p>These are downloaded to a local cache (use the internal function <code>cellxgenedp:::.cellxgenedb_cache_path()</code> for the location of the cache), so the process is only time-consuming the first time.</p>
<p><code>H5AD</code> files can be converted to <em>R</em> / <em>Bioconductor</em> objects using the <a href="https://bioconductor.org/packages/zelkonverter" class="external-link">zellkonverter</a> package.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h5ad</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/zellkonverter/man/readH5AD.html" class="external-link">readH5AD</a></span><span class="op">(</span><span class="va">local_file</span>, use_hdf5 <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; + '/home/runner/.cache/R/basilisk/1.10.2/0/bin/conda' 'create' '--yes' '--prefix' '/home/runner/.cache/R/basilisk/1.10.2/zellkonverter/1.8.0/zellkonverterAnnDataEnv-0.8.0' 'python=3.8.13' '--quiet' '-c' 'conda-forge'</span></span>
<span><span class="co">#&gt; + '/home/runner/.cache/R/basilisk/1.10.2/0/bin/conda' 'install' '--yes' '--prefix' '/home/runner/.cache/R/basilisk/1.10.2/zellkonverter/1.8.0/zellkonverterAnnDataEnv-0.8.0' 'python=3.8.13'</span></span>
<span><span class="co">#&gt; + '/home/runner/.cache/R/basilisk/1.10.2/0/bin/conda' 'install' '--yes' '--prefix' '/home/runner/.cache/R/basilisk/1.10.2/zellkonverter/1.8.0/zellkonverterAnnDataEnv-0.8.0' '-c' 'conda-forge' 'python=3.8.13' 'anndata=0.8.0' 'h5py=3.6.0' 'hdf5=1.12.1' 'natsort=8.1.0' 'numpy=1.22.3' 'packaging=21.3' 'pandas=1.4.2' 'python=3.8.13' 'scipy=1.7.3' 'sqlite=3.38.2'</span></span>
<span><span class="co">#&gt; Warning: 'X' matrix does not support transposition and has been skipped</span></span>
<span><span class="va">h5ad</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 33234 31696 </span></span>
<span><span class="co">#&gt; metadata(3): default_embedding schema_version title</span></span>
<span><span class="co">#&gt; assays(1): X</span></span>
<span><span class="co">#&gt; rownames(33234): ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span></span>
<span><span class="co">#&gt;   ENSG00000268674</span></span>
<span><span class="co">#&gt; rowData names(4): feature_is_filtered feature_name feature_reference</span></span>
<span><span class="co">#&gt;   feature_biotype</span></span>
<span><span class="co">#&gt; colnames(31696): CMGpool_AAACCCAAGGACAACC CMGpool_AAACCCACAATCTCTT ...</span></span>
<span><span class="co">#&gt;   K109064_TTTGTTGGTTGCATCA K109064_TTTGTTGGTTGGACCC</span></span>
<span><span class="co">#&gt; colData names(34): donor_id self_reported_ethnicity_ontology_term_id</span></span>
<span><span class="co">#&gt;   ... self_reported_ethnicity development_stage</span></span>
<span><span class="co">#&gt; reducedDimNames(3): X_pca X_tsne X_umap</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
<p>The <code>SingleCellExperiment</code> object is a matrix-like object with rows corresponding to genes and columns to cells. Thus we can easily explore the cells present in the data.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h5ad</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">h5ad</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">sex</span>, <span class="va">donor_id</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 7 × 3</span></span>
<span><span class="co">#&gt;   sex    donor_id                     n</span></span>
<span><span class="co">#&gt;   &lt;fct&gt;  &lt;fct&gt;                    &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1 female D1                        2303</span></span>
<span><span class="co">#&gt; 2 female D2                         864</span></span>
<span><span class="co">#&gt; 3 female D3                        2517</span></span>
<span><span class="co">#&gt; 4 female D4                        1771</span></span>
<span><span class="co">#&gt; 5 female D5                        2244</span></span>
<span><span class="co">#&gt; 6 female D11                       7454</span></span>
<span><span class="co">#&gt; 7 female pooled [D9,D7,D8,D10,D6] 14543</span></span></code></pre></div>
</div>
<div class="section level1">
<h1 id="next-steps">Next steps<a class="anchor" aria-label="anchor" href="#next-steps"></a>
</h1>
<p>The <a href="https://bioconductor.org/books/OSCA" class="external-link">Orchestrating Single-Cell Analysis with Bioconductor</a> online resource provides an excellent introduction to analysis and visualization of single-cell data in <em>R</em> / <em>Bioconductor</em>. Extensive opportunities for working with AnnData objects in <em>R</em> but using the native python interface are briefly described in, e.g., <code><a href="https://rdrr.io/pkg/zellkonverter/man/AnnData-Conversion.html" class="external-link">?AnnData2SCE</a></code> help page of <a href="https://bioconductor.org/packages/zelkonverter" class="external-link">zellkonverter</a>.</p>
<p>The <a href="https://bioconductor.org/packages/hca" class="external-link">hca</a> package provides programmatic access to the Human Cell Atlas <a href="https://data.humancellatlas.org/explore" class="external-link">data portal</a>, allowing retrieval of primary as well as derived single-cell data files.</p>
</div>
<div class="section level1">
<h1 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h1>
<div class="sourceCode" id="cb25"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; R version 4.2.2 (2022-10-31)</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running under: Ubuntu 22.04.1 LTS</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; locale:</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] stats4    stats     graphics  grDevices datasets  utils     methods  </span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [8] base     </span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] cellxgenedp_1.3.3           dplyr_1.1.0                </span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [3] SingleCellExperiment_1.20.0 SummarizedExperiment_1.28.0</span></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] Biobase_2.58.0              GenomicRanges_1.50.2       </span></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] GenomeInfoDb_1.34.9         IRanges_2.32.0             </span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] S4Vectors_0.36.1            BiocGenerics_0.44.0        </span></span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] MatrixGenerics_1.10.0       matrixStats_0.63.0         </span></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] zellkonverter_1.8.0        </span></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] Rcpp_1.0.10            here_1.0.1             dir.expiry_1.6.0      </span></span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [4] lattice_0.20-45        png_0.1-8              rprojroot_2.0.3       </span></span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] digest_0.6.31          utf8_1.2.3             mime_0.12             </span></span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [10] R6_2.5.1               evaluate_0.20          httr_1.4.4            </span></span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] pillar_1.8.1           basilisk_1.10.2        zlibbioc_1.44.0       </span></span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [16] rlang_1.0.6            curl_5.0.0             Matrix_1.5-1          </span></span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [19] DT_0.27                reticulate_1.28        rmarkdown_2.20        </span></span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [22] htmlwidgets_1.6.1      RCurl_1.98-1.10        shiny_1.7.4           </span></span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [25] DelayedArray_0.24.0    compiler_4.2.2         httpuv_1.6.8          </span></span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [28] xfun_0.37              pkgconfig_2.0.3        htmltools_0.5.4       </span></span>
<span id="cb25-39"><a href="#cb25-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [31] tidyselect_1.2.0       tibble_3.1.8           GenomeInfoDbData_1.2.9</span></span>
<span id="cb25-40"><a href="#cb25-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [34] codetools_0.2-18       fansi_1.0.4            withr_2.5.0           </span></span>
<span id="cb25-41"><a href="#cb25-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [37] later_1.3.0            bitops_1.0-7           rjsoncons_1.0.0       </span></span>
<span id="cb25-42"><a href="#cb25-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [40] basilisk.utils_1.10.0  grid_4.2.2             jsonlite_1.8.4        </span></span>
<span id="cb25-43"><a href="#cb25-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [43] xtable_1.8-4           lifecycle_1.0.3        magrittr_2.0.3        </span></span>
<span id="cb25-44"><a href="#cb25-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [46] cli_3.6.0              XVector_0.38.0         renv_0.16.0           </span></span>
<span id="cb25-45"><a href="#cb25-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [49] promises_1.2.0.1       ellipsis_0.3.2         filelock_1.0.2        </span></span>
<span id="cb25-46"><a href="#cb25-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [52] generics_0.1.3         vctrs_0.5.2            tools_4.2.2           </span></span>
<span id="cb25-47"><a href="#cb25-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [55] glue_1.6.2             parallel_4.2.2         fastmap_1.1.0         </span></span>
<span id="cb25-48"><a href="#cb25-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [58] yaml_2.3.7             knitr_1.42</span></span></code></pre></div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/cellxgenedp" class="external-link">View on Bioconductor</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0" class="external-link">Artistic-2.0</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing cellxgenedp</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Martin Morgan <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-5874-8148" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Kayla Interdonato <br><small class="roles"> Author </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Martin Morgan, Kayla Interdonato.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
